polkadot/node/network/statement-distribution/src/v2/requests.rs

```diff
@@ -462,10 +478,17 @@ fn find_request_target_with_update(
 	candidate_identifier: &CandidateIdentifier,
 	props: &RequestProperties,
 	peer_advertised: impl Fn(&CandidateIdentifier, &PeerId) -> Option<StatementFilter>,
+	response_manager: &ResponseManager,
 ) -> Option<PeerId> {
 	let mut prune = Vec::new();
 	let mut target = None;
 	for (i, p) in known_by.iter().enumerate() {
+
+		// If we are already sending to that peer, skip for now
+		if response_manager.is_sending_to(p) {
```

Comment:
I would investigate here, if this might create more Nones, than before, I'm under impression that we call this functions that aren't that many peers which advertised the candidates to us.

Also, if we don't expect this to  happen let's sprinkle with warnings, so that we know if we hit this conditions in the wild.

Reply:
> I would investigate here, if this might create more Nones, than before

It will definitely create more `None`s. How much more I'm not certain, but generally I don't think that those are bad because it means that we are already downloading other data from that same peer so might as well finish that before starting another. Additionally it means that we will prioritise another node with that data so it might even help with load balancing.

> Also, if we don't expect this to happen let's sprinkle with warnings, so that we know if we hit this conditions in the wild.

We do expect this to happen. When we iterate over the peers this is doing the filtering (removes peers we are already sending to). So in the wild that might happen somewhat frequently especially with async backing.